src/bin/tikv-fail.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(plugin)]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+#![cfg_attr(not(feature = "dev"), allow(unknown_lints))]
+#![allow(needless_pass_by_value)]
+
+/// Inject failures to `TikV`.
+///
+/// TODO: Integrate into tikv-ctl.
+///
+/// # Examples
+///
+/// ```sh
+/// ./tikv-fail -a 127.0.0.1:9090 inject\
+//      -f tikv::raftstore::store::store::raft_between_save\
+///     -a "panic(fail_point_raft_between_write)"
+/// ```
+///
+
+extern crate tikv;
+extern crate clap;
+extern crate grpcio as grpc;
+extern crate protobuf;
+extern crate kvproto;
+
+use std::str;
+use std::time::Duration;
+use std::sync::Arc;
+
+use clap::{App, Arg, SubCommand};
+use grpc::{CallOption, ChannelBuilder, EnvBuilder};
+use protobuf::RepeatedField;
+use kvproto::debugpb;
+use kvproto::debugpb_grpc::DebugClient;
+
+fn main() {
```

Comment:
Can this be merged with TiKV Control?

Reply:
Yes, it has been added to TODO. https://github.com/pingcap/tikv/pull/2354/files#diff-3a911291a9738b9e1ccccbde951efdbcR21

Currently, @hicqu is refactoring tikv-ctl.